### PR TITLE
FAI-1803: Add ServiceNow source/converter

### PR DESCRIPTION
**Pull request by willmarks**
## Description

Add source and converter for pulling Incidents and Users from ServiceNow.

## Notes

- I am still trying to find a way to use anything other than basic auth for API access...
- There does not appear to be any notion of a service linked to an incident (as far as I can tell...)

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

**Comment by cjwooo** [comment]
@willmarks  Do you have a config.json we can use for source acceptance testing? If so, can you send it to me

**Comment by willmarks** [comment]
I do not have one for the acceptance tests as the instance I was using for testing is a developer instance that hibernates whenever not in use and will be reclaimed after not being used for 10 days. Any tests pointed to it would start failing as soon as the instance hibernates.

File: sources/servicenow-source/src/streams/incidents.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    protected readonly logger: AirbyteLogger
+  ) {
+    super(logger);
+  }
+
+  getJsonSchema(): Dictionary<any, string> {
+    return require('../../resources/schemas/incident.json');
+  }
+  get primaryKey(): StreamKey {
+    return ['number.value'];
```
Comment (cjwooo) [inline sources/servicenow-source/src/streams/incidents.ts:28]: Change to `['number', 'value']` here and throughout (catalog jsons)

File: sources/servicenow-source/src/streams/incidents.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  }
+  get cursorField(): string | string[] {
+    return ['sys_updated_on.value'];
```
Comment (cjwooo) [inline sources/servicenow-source/src/streams/incidents.ts:31]: Change to `['sys_updated_on', 'value']` here and throughout

File: sources/servicenow-source/acceptance-test-config.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      configured_catalog_path: "test_files/full_configured_catalog.json"
+  full_refresh:
+    - config_path: "test_files/config.json"
+      configured_catalog_path: "test_files/full_configured_catalog.json"
+  incremental:
+    - config_path: "test_files/config.json"
+      configured_catalog_path: "test_files/incremental_configured_catalog.json"
+      future_state_path: "test_files/abnormal_state.json"
+      cursor_paths:
+        incidents: [ "sys_updated_on.value" ]
```
Comment (cjwooo) [inline sources/servicenow-source/acceptance-test-config.yml:24]: I know we're not running these tests for real but the cursor paths here should be the key in the resulting state object, not the cursor field specified by the stream in the catalog. This enables the test container to read the correct state key.

File: sources/servicenow-source/resources/spec.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      "url"
+    ],
+    "additionalProperties": false,
+    "properties": {
+      "username": {
+        "order": 0,
+        "type": "string",
+        "title": "ServiceNow Username",
+        "description": "The users should have the roles: sn_incident_read, cmdb_query_builder_read",
+        "airbyte_secret": true
```
Comment (cjwooo) [inline sources/servicenow-source/resources/spec.json:19]: Does the username need to be secret?

File: sources/servicenow-source/src/streams/incidents.ts (hunk 1 above)
Comment (cjwooo) [reply sources/servicenow-source/src/streams/incidents.ts:28]: This is how to represent nested fields in the resulting stream catalog

**Review by cjwooo — APPROVED** [review]

File: sources/servicenow-source/acceptance-test-config.yml (hunk 3 above)
Comment (willmarks) [reply sources/servicenow-source/acceptance-test-config.yml]: Ahh. ok that makes sense, thanks!

**Comment by tovbinm** [comment]
@willmarks please add it to the README + catalogs

**Pull request merged**